modules/core/src/stat.cpp

```diff
@@ -2181,115 +2216,120 @@ static bool ocl_minMaxIdx( InputArray _src, double* minVal, double* maxVal, int*
+            type == CV_16UC1 ? (ippiMaskMinMaxIndxFuncC1)ippiMinMaxIndx_16u_C1MR :
+            type == CV_32FC1 ? (ippiMaskMinMaxIndxFuncC1)ippiMinMaxIndx_32f_C1MR : 0;
+        CV_SUPPRESS_DEPRECATED_END
 
-                if( ippFuncC1 )
+        if( ippFuncC1 )
+        {
+            Ipp32f min, max;
+            IppiPoint minp, maxp;
+            if( ippFuncC1(src.ptr(), (int)src.step[0], mask.ptr(), (int)mask.step[0], sz, &min, &max, &minp, &maxp) >= 0 )
+            {
+                if( minVal )
+                    *minVal = (double)min;
+                if( maxVal )
+                    *maxVal = (double)max;
+                if( !minp.x && !minp.y && !maxp.x && !maxp.y && !mask.ptr()[0] )
+                    minp.x = maxp.x = -1;
+                if( minIdx )
                 {
-                    Ipp32f min, max;
-                    IppiPoint minp, maxp;
-                    if( ippFuncC1(src.ptr(), (int)src.step[0], mask.ptr(), (int)mask.step[0], sz, &min, &max, &minp, &maxp) >= 0 )
-                    {
-                        if( minVal )
-                            *minVal = (double)min;
-                        if( maxVal )
-                            *maxVal = (double)max;
-                        if( !minp.x && !minp.y && !maxp.x && !maxp.y && !mask.ptr()[0] )
-                            minp.x = maxp.x = -1;
-                        if( minIdx )
-                        {
-                            size_t minidx = minp.y * cols + minp.x + 1;
-                            ofs2idx(src, minidx, minIdx);
-                        }
-                        if( maxIdx )
-                        {
-                            size_t maxidx = maxp.y * cols + maxp.x + 1;
-                            ofs2idx(src, maxidx, maxIdx);
-                        }
-                        CV_IMPL_ADD(CV_IMPL_IPP);
-                        return;
-                    }
-                    setIppErrorStatus();
+                    size_t minidx = minp.y * cols + minp.x + 1;
+                    ofs2idx(src, minidx, minIdx);
+                }
+                if( maxIdx )
+                {
+                    size_t maxidx = maxp.y * cols + maxp.x + 1;
+                    ofs2idx(src, maxidx, maxIdx);
                 }
+                return true;
             }
-            else
-            {
-                typedef IppStatus (CV_STDCALL* ippiMinMaxIndxFuncC1)(const void *, int, IppiSize, Ipp32f *, Ipp32f *, IppiPoint *, IppiPoint *);
-
-                CV_SUPPRESS_DEPRECATED_START
-                ippiMinMaxIndxFuncC1 ippFuncC1 =
-                    depth == CV_8U ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_8u_C1R :
-                    depth == CV_8S ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_8s_C1R :
-                    depth == CV_16U ? (ippiMinMaxIndxFuncC1)ippiMinMaxIndx_16u_C1R :
-                #if !((defined _MSC_VER && defined _M_IX86) || defined __i386__)
```

Comment:
This condition should be restored.

http://pullrequest.opencv.org/buildbot/builders/precommit_windows32/builds/196/steps/test_imgproc_1/logs/stdio
(this buildslave's CPU is AMD A10-6800K, so something went wrong in this case)